### PR TITLE
[feature](Cloud) Only cloud mode and instance with enable_storage_vault can use storage vault stmt

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateStorageVaultStmt.java [hunk 1]
```diff
@@ -72,6 +74,12 @@ public void setStorageVaultType(StorageVault.StorageVaultType type) throws UserE
 
     @Override
     public void analyze(Analyzer analyzer) throws UserException {
+        if (!Config.isCloudMode()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateStorageVaultStmt.java:77]:
```suggestion
        if (Config.isNotCloudMode()) {
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateStorageVaultStmt.java [hunk 2]
```diff
@@ -72,6 +74,12 @@ public void setStorageVaultType(StorageVault.StorageVaultType type) throws UserE
...
+            throw new AnalysisException("Storage Vault is only supported for cloud mode");
+        }
+        if (!((CloudEnv) Env.getCurrentEnv()).getEnableStorageVault()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateStorageVaultStmt.java:80]: Add a comment: Some s3 back-ended storage does need to use storage vault.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SetDefaultStorageVaultStmt.java [hunk 3]
```diff
@@ -29,6 +35,17 @@ public String getStorageVaultName() {
         return vaultName;
     }
 
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
+        if (!Config.isCloudMode()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SetDefaultStorageVaultStmt.java:40]: ditto

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run feut

**Comment by ByteYue** [comment]
run feut

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**